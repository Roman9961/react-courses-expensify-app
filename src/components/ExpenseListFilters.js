import React from 'react';
import {connect} from 'react-redux';
import {DateRangePicker} from  'react-dates';
import {setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate} from '../actions/filters';

class ExpenseListFilters extends React.Component{
    state = {
        calendarFocused: null
    };
    onDatesChange = ({startDate, endDate}) => {
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate));
    };
    onFocusChange = (calendarFocused) => {
       this.setState(()=>({calendarFocused}));
    };
    render(){
        return (
            <React.Fragment>
                <input type="text" value={this.props.filters.text} onChange={(e)=>{
                    this.props.dispatch(setTextFilter(e.target.value));
                }}/>
                <select value={this.props.filters.sortBy}
                        onChange={(e)=>{
                            e.target.value==='date'?this.props.dispatch(sortByDate()):this.props.dispatch(sortByAmount());
                        }}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
                    endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
                    onDatesChange={this.onDatesChange} // PropTypes.func.isRequired,
                    focusedInput={this.state.calendarFocused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
                    onFocusChange={this.onFocusChange} // PropTypes.func.isRequired,
                    showClearDates={true}
                    numberOfMonths = {1}
                    isOutsideRange ={()=>false}
                />

            </React.Fragment>
        )
    }
}



const mapStateToProps = (state) =>{
    return{
        filters: state.filters
    }
}

export default connect(mapStateToProps)(ExpenseListFilters);