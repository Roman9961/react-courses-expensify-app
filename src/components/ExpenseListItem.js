import React from 'react';
import { connect } from 'react-redux';
import {NavLink} from 'react-router-dom';

export const ExpenseListItem = ({id, description, amount, createdAt}) => (
    <React.Fragment>
    <NavLink to={`/edit/${id}`}>
        <h3>{description}</h3>
    </NavLink>
       <p>{amount}-{createdAt}</p>
    </React.Fragment>
);


export default connect()(ExpenseListItem);