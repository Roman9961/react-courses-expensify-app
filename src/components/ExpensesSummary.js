import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';


export const ExpensesSummary = (props) => (
    <div>
        <p>Viewing {props.expenseCount} expenses totaling {numeral(props.expenseTotal /100).format('$0,0.00')}</p>
    </div>
);

const mapStateToProps = (state)=>{
    const selectedExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: selectedExpenses.length,
        expenseTotal: getExpensesTotal(selectedExpenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary);
