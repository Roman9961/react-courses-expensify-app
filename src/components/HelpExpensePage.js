import React from 'react';

const HelpExpensePage = () => (
    <p>Help</p>
);

export default HelpExpensePage;