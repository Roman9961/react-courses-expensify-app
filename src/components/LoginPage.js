import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from  '../actions/auth';

export const LoginPage = ({startLogin}) => (
    <p>
        <button onClick={startLogin}>Login</button>
    </p>
);

const mapDispatchToProps = (dispatch) => {
    return {
        startLogin: ()=> dispatch(startLogin())
    }
};

export default connect(undefined, mapDispatchToProps)(LoginPage);
