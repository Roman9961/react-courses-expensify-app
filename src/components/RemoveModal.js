import React from 'react';
import { connect } from 'react-redux';
import Modal from 'react-modal';
import { startRemoveExpense} from '../actions/expenses';
import { resolveRemoveModal } from '../actions/modals';

Modal.setAppElement('#app');


const RemoveModal = (props) => {
    const handleCloseModal = ()=> {
        props.resolveRemoveModal(false);
    };
    const handleRemoveExpense = ()=> {
        props.resolveRemoveModal(false);
        props.startRemoveExpense({id: props.expense.id});
        props.history.push('/');
    };
    return (
        <Modal
            isOpen={props.removeModal}
            onRequestClose={handleCloseModal}
            contentLabel="Error"
            closeTimeoutMS={200}
            className="modal"
        >
            <h3 className="modal__title modal__title--error">Are You Shure?</h3>
            {props.removeModal && <p className="modal__body modal__body--error">You trying to delete this expense!</p>}
            <div className="modal-buttons">
            <button className="button button--modal" onClick={handleRemoveExpense}>Ok</button>
            <button className="button button--modal" onClick={handleCloseModal}>Cancel</button>
            </div>
        </Modal>
    )
};

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find( (expense) => expense.id === props.match.params.id ),
        removeModal:  state.removeModal
    }
};
const mapDispatchToProps = (dispatch) => {
    return {
        startRemoveExpense: (data) => dispatch(startRemoveExpense(data)),
        resolveRemoveModal: (data) => dispatch(resolveRemoveModal(data))
    }
};
export default connect(mapStateToProps, mapDispatchToProps)(RemoveModal);
