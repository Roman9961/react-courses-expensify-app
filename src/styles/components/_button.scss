.button{
  background-color: dodgerblue;
  border: none;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  padding: $s-size;
  text-align: center;
  text-decoration: none;
  line-height: 1;
}
.button--link{
  background: none;
}
.button--remove{
  background: #888888;
}

.button--modal{
  min-width: 100px;
}