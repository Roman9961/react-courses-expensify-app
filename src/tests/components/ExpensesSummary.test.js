import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('Should render ExpensesSummary correctly', ()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount = {2} expenseTotal={2312312}/>);
    expect(wrapper).toMatchSnapshot();
});


