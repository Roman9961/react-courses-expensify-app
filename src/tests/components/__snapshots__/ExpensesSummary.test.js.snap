// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary correctly 1`] = `
<div>
  <p>
    Viewing 
    2
     expenses totaling 
    $23,123.12
  </p>
</div>
`;
